docs/src/main/asciidoc/spring-security.adoc

```diff
@@ -0,0 +1,252 @@
+
+For more information about `security-properties` you can check the guide of link:security-properties[quarkus-elytron-security-properties-file] extension.
+
+== GreetingController
+
+The Quarkus maven plugin automatically generated a controller with the Spring Web annotations to define our REST endpoint (instead of the JAX-RS ones used by default)
+The `src/main/java/org/acme/spring/web/GreetingController.java` file looks as follows:
+
+[source,java]
+----
+package org.acme.spring.web;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@RestController
+@RequestMapping("/greeting")
+public class GreetingController {
+
+    @GetMapping
+    public String hello() {
+        return "hello";
+    }
+}
+----
+
+== GreetingControllerTest
+
+Note that a test for the controller has been created as well:
+
+[source, java]
+----
+package org.acme.spring.web;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+public class GreetingControllerTest {
+
+    @Test
+    public void testHelloEndpoint() {
+        given()
+          .when().get("/greeting")
+          .then()
+             .statusCode(200)
+             .body(is("hello"));
+    }
+
+}
+----
+
+== Package and run the application
+
+Run the application with: `./mvnw compile quarkus:dev`.
+Open your browser to http://localhost:8080/greeting.
+
+The result should be: `{"message": "hello"}`.
+
+== Modify the controller to secure the `hello` method
+
+Since we are going to be using Spring Security annotations to define the authorizations to execute the `hello` method, you can add the Spring `@Secured` annotation allowing secure the resources by restricting which roles are able to execute it. The controller above will look as follows:
+
+[source,java]
+----
+package org.acme.spring.web;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@RestController
+@RequestMapping("/greeting")
+public class GreetingController {
+
+    @Secured("admin")
+    @GetMapping
+    public String hello() {
+        return "hello";
+    }
+}
+----
+
+We use Quarkus `security-properties` extension to load the authentication information. It uses the main `application.properties` Quarkus configuration file to embed this information. Please, for more information about this extension check link:the assoiated guide[quarkus-elytron-security-properties-file].
```

Comment:
I propose to rephrase this paragraph as: 

```
The easiest way to setup users and roles for our example is to use the `security-properties` extension. This extensions essentially allows users and roles to be defined in properties files whose names are configured from the main Quarkus configuration file - `application.properties`
```

Reply:
s/setup/set up/